src/net.cpp

```diff
@@ -1805,11 +1807,24 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             if (nTries > 100)
                 break;
 
-            CAddrInfo addr = addrman.SelectTriedCollision();
+            CAddress addr;
+            if (!m_anchors.empty() && (nOutboundBlockRelay < m_max_outbound_block_relay) && !fFeeler) {
```

Comment:
The order is not inverted. The first target "anchors" is added, and was not present before.
If anchors would not the first, they could be connected as other type nodes, and anchoring could not happen.

Reply:
But anchors nodes are block-only, so now you have block-only-as-anchors, full-relay, block-only-classics. You can still test addresses returned from `addrman` against `m_anchors` andon continue in case of equality to avoid the problem you're describing, but don't think that changes something, do we favor first-connected nodes in anyway ?